python/grass/jupyter/region.py

```diff
@@ -215,6 +215,86 @@ def set_region_from_command(self, module, **kwargs):
             return
 
 
+class RegionManagerForSeries:
+    """Region manager for SeriesMap"""
+
+    def __init__(self, use_region, saved_region, width, height, env):
+        """Manages region during rendering.
+
+        :param use_region: if True, use either current or provided saved region,
+                          else derive region from rendered layers
+        :param saved_region: if name of saved_region is provided,
+                            this region is then used for rendering
+        :param width: rendering width
+        :param height: rendering height
+        :param env: environment for rendering
+        """
+        self._env = env
+        self._width = width
+        self._height = height
+        self._use_region = use_region
+        self._saved_region = saved_region
+        self._extent_set = False
+        self._resolution_set = False
+
+    def set_region_from_rasters(self, rasters):
+        """Sets computational region for rendering from a series of rasters.
+
+        This function sets the region from a series of rasters. If the extent or resolution has
+        already been set by calling this function previously or by the set_region_from_vectors()
+        function, this function will not modify it.
+
+        If user specified the name of saved region during object's initialization,
+        the provided region is used. If it's not specified
+        and use_region=True, current region is used.
+        """
+        if self._saved_region:
+            self._env["GRASS_REGION"] = gs.region_env(
+                region=self._saved_region, env=self._env
+            )
+            return
+        if self._use_region:
+            # use current
+            return
+        if self._resolution_set and self._extent_set:
+            return
+        try:
+            if not self._resolution_set and not self._extent_set:
+                self._env["GRASS_REGION"] = gs.region_env(raster=rasters, env=self._env)
+                self._extent_set = True
+                self._resolution_set = True
+            elif not self._resolution_set:
+                self._env["GRASS_REGION"] = gs.region_env(
+                    align=rasters[0], env=self._env
+                )
+                self._resolution_set = True
+        except CalledModuleError:
```

Comment:
Any specific reason for catching CalledModuleError here?

Reply:
No, I think I just included it because it's necessary in other `set_region_from_...` functions and I copied/pasted for the template of this function. I've removed it.